resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/tpr/sparkJobResourceController.scala

```diff
@@ -0,0 +1,316 @@
+  }
+
+  /**
+   * This method Patches in Batch or singly a resource of kind "SparkJob" with the specified name
+   * @return Unit
+   * @throws IOException Due to failure from execute call
+   * @throws SparkException when PATCH request is unsuccessful
+   */
+  override def updateJobObject(jobResourcePatches: Seq[JobResourcePatch]): Unit = {
+    val payload = jobResourcePatches map { jobResourcePatch =>
+        ("op" -> "replace") ~
+          ("path" -> jobResourcePatch.fieldPath) ~
+          ("value" -> jobResourcePatch.value.toString)
+    }
+    val requestBody =
+      RequestBody.create(
+        MediaType.parse("application/json-patch+json"),
+        compact(render(payload)))
+    val requestPathSegments = Seq(
+      "apis", TPR_API_GROUP, TPR_API_VERSION, "namespaces",
+      namespace, "sparkjobs", jobResourcePatches.head.name)
+    val url = generateHttpUrl(requestPathSegments)
+    val request = new Request.Builder()
+      .patch(requestBody)
+      .url(url)
+      .build()
+
+    logDebug(s"Update Request: $request")
+    var response: Response = null
+    try {
+      response = httpClient.newCall(request).execute()
+    } catch {
+      case x: IOException =>
+        val msg =
+          s"Failed to get resource ${jobResourcePatches.head.name}. ${x.getMessage}."
+        logError(msg)
+        response.close()
+        throw new SparkException(msg)
+    }
+    completeRequestWithExceptionIfNotSuccessful(
+      "patch",
+      response,
+      Option(Seq(jobResourcePatches.head.name, response.message(), compact(render(payload))))
+    )
+
+    response.body().close()
+    logDebug(s"Successfully patched resource ${jobResourcePatches.head.name}.")
+  }
+
+  private def generateHttpUrl(urlSegments: Seq[String],
+    querySegments: Seq[(String, String)] = Seq.empty[(String, String)]): HttpUrl = {
+
+    val urlBuilder = HttpUrl.parse(kubeMaster).newBuilder
+    urlSegments map { pathSegment =>
+      urlBuilder.addPathSegment(pathSegment)
+    }
+    querySegments map {
+      case (query, value) => urlBuilder.addQueryParameter(query, value)
+    }
+    urlBuilder.build()
+  }
+
+  private def completeRequestWithExceptionIfNotSuccessful(
+    requestType: String,
+    response: Response,
+    additionalInfo: Option[Seq[String]] = None): Unit = {
+
+    if (!response.isSuccessful) {
+      response.body().close()
+      val msg = new ArrayBuffer[String]
+      msg += s"Failed to $requestType resource."
+
+      additionalInfo match {
+        case Some(info) =>
+          for (extraMsg <- info) {
+            msg += extraMsg
+          }
+        case None =>
+      }
+
+      val finalMessage = msg.mkString(" ")
+      logError(finalMessage)
+      throw new SparkException(finalMessage)
+    }
+  }
+
+  def extractHttpClientFromK8sClient(client: BaseClient): OkHttpClient = {
+    val field = classOf[BaseClient].getDeclaredField("httpClient")
+    try {
+      field.setAccessible(true)
```

Comment:
I'm looking at `BaseClient` and it has a `getHttpClient` method. Is it therefore necessary to use reflection here?
https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/BaseClient.java#L113